Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.cc

```diff
@@ -48,12 +51,52 @@ using model::SnapshotVersion;
 using nanopb::ByteString;
 using nanopb::Message;
 using nanopb::StringReader;
+using util::BackgroundQueue;
+using util::Executor;
+
+/**
+ * An accumulator for results produced asynchronously. This accumulates
+ * values in a vector to avoid contention caused by accumulating into more
+ * complex structures like immutable::SortedMap.
+ */
+template <typename T>
+class AsyncResults {
```

Comment:
Also, please wrap this class in an unnamed namespace (wrap the type aliases above too).